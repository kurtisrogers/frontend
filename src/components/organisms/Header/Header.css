header {
  background: var(--brand-black);
}

header .logo {
  text-transform: uppercase;
  text-decoration: none;
  outline: 2px solid var(--brand-black);
  padding: .5rem .75rem;
}

header .logo:focus-visible {
  outline-color: var(--brand-white);
  color: var(--brand-black);
  background: var(--a11y-focus-yellow);
}

header > .content-skipper {
  position: absolute;
  display: inline-block;
  inset: -999999px auto auto -999999px;
  font-size: 0px;
}

header > .content-skipper:focus-visible {
  position: relative;
  inset: 0px;
  font-size: 1em;
  margin: .5rem;
}

header > .content {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  max-width: var(--brand-base-grid);
  margin: 0 auto;
}