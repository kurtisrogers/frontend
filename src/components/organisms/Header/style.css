header {
  width: 100%;
  z-index: 20;
  position: -webkit-sticky;
  position: sticky;
  top: -1px;
}

header > .content {
  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: var(--brand-base-grid);
  margin: 1rem;
  background: var(--brand-white);
  padding: 0.5rem 2rem 0.5rem 0.5rem;
}

header nav.site-navigation > ul {
  display: flex;
  padding: 0px;
}

header nav.site-navigation li {
  list-style-type: none;
}

header nav.site-navigation a {
  padding: .5rem .75rem;
  margin-left: .5rem;
  outline: 2px solid var(--brand-white);
  color: var(--brand-black);
}

header nav.site-navigation a:hover {
  text-decoration: underline;
}

@media screen and (min-width: 1320px) {
  header > .content {
    width: 100%;
    margin: 1rem auto;
  }
}
