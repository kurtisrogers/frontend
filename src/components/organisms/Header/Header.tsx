import { A } from "@solidjs/router";
import Navigation from "@/components/molecules/Navigation/Navigation";
import "./Header.css";

type Item = {
  link: string;
  text: string;
}

type Props = {
  menuItems?: Item[];
  children?: any;
}

export default function Header( props: Readonly<Props> ) {
  return (
    <header>
      <A href="#main-content" class="button content-skipper">Skip to main content</A>
      <div class="content">
        <A href="/" class="logo text-white"><strong>K</strong>urtis <strong>R</strong>ogers</A>
        <div>
          <Navigation />
          {props.children}
        </div>
      </div>
    </header>
  )
}
