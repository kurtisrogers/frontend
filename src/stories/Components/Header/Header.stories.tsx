import type { Meta, StoryObj } from 'storybook-solidjs';
import Header from '@/components/organisms/Header/Header';
import { Router, Route } from '@solidjs/router';

export default {
  title: 'Example/Header',
  component: Header,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/solid/writing-docs/docs-page
  tags: ['autodocs'],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/solid/configure/story-layout
    layout: 'fullscreen',
  },
  args: {
    title: 'something',
  },
}

const Template = (args: any) => (
  <Router>
    <Header {...args} />
  </Router>
);

export const Default = Template.bind({})
Default.args = {
  title: 'something',
} as any;
