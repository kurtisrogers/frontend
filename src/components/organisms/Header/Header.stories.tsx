import Header from "@/components/organisms/Header";
import "./style.css";

export default {
  title: "Example/Header",
  component: Header,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/solid/writing-docs/docs-page
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/solid/configure/story-layout
    layout: "fullscreen"
  }
};

const Template = () => {
  return (
    <header>
      <a href="#main-content" class="button content-skipper">
        Skip to main content
      </a>
      <div class="content">
        <a href="/" class="logo text-white">
          <strong>Kurtis Rogers</strong>
        </a>
        <div>
          <nav class="site-navigation">
            <ul>
              <li>
                <a href="/">Home</a>
                <a href="/">About</a>
                <a href="/">Work</a>
                <a href="/">Contact</a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

export const Default = Template.bind({});
// Default.args = {
//   // Add any props you want to pass to the Header component
// };
