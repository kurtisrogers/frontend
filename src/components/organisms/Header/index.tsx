import { JSX } from "solid-js";
import Navigation from "@/components/molecules/Navigation";
import SkipLink from "@/components/atoms/Skiplink";
import Logo from "@/components/atoms/Logo";
import "./style.css";

type Item = {
  link: string;
  text: string;
};

type Props = {
  menuItems?: Item[];
  children?: JSX.Element[];
};

export default function Header({ children }: Readonly<Props>) {
  return (
    <>
      <header tabindex="-1">
        <SkipLink id="main" name="main content" isTag={true} visibleOnFocusOnly={true} />
        <div class="content">
          <Logo />
          <div>
            <Navigation classes={"site-navigation"} />
            {children}
          </div>
        </div>
      </header>
    </>
  );
}
