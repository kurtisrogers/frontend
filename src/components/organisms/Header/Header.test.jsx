// src/components/LinkButton.test.jsx
import { render } from '@solidjs/testing-library';
import { Router } from '@solidjs/router';
import Header from '.';
import { describe, it, expect } from 'vitest';

vi.mock(import("@solidjs/router"), async (importOriginal) => {
  const actual = await importOriginal()
  return {
    ...actual,
    A: (props) => <a {...props} data-testid="mock-link">{props.children}</a>
  }
})

describe('<Header />', () => {
  it('renders correctly and links to the correct route', () => {
    const { getByText } = render(() => <Header />);
    const linkButton = getByText('loading');

    expect(linkButton).toMatchSnapshot()
    expect(linkButton).toBeInTheDocument();
    expect(linkButton).toHaveAttribute('href', '/');
  });
});
