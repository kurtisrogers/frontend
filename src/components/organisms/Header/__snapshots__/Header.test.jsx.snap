// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > renders correctly and links to the correct route 1`] = `
<a
  activeclass="current"
  data-testid="mock-link"
  href="/"
>
  loading
</a>
`;
